File: medplum-java/medplum-java-model/src/main/java/com/fhir/NutritionOrder_OralDiet.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableNutritionOrder_OralDiet.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface NutritionOrder_OralDiet {
  Optional<List<Extension>> modifierExtension();

  Optional<String> instruction();

  Optional<List<Timing>> schedule();

  Optional<List<NutritionOrder_Nutrient>> nutrient();

  Optional<List<NutritionOrder_Texture>> texture();

  Optional<List<CodeableConcept>> type();

  Optional<List<CodeableConcept>> fluidConsistencyType();

  Optional<List<Extension>> extension();

  Optional<String> id();

  static ImmutableNutritionOrder_OralDiet.Builder builder() {
    return ImmutableNutritionOrder_OralDiet.builder();
  }
}

File: medplum-typescript/src/model/fhir/NutritionOrder_OralDiet.ts
import { CodeableConcept } from "./CodeableConcept";
import { Extension } from "./Extension";
import { NutritionOrder_Nutrient } from "./NutritionOrder_Nutrient";
import { NutritionOrder_Texture } from "./NutritionOrder_Texture";
import { Timing } from "./Timing";

export interface NutritionOrder_OralDiet {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    type: CodeableConcept[] | null | undefined;
    schedule: Timing[] | null | undefined;
    nutrient: NutritionOrder_Nutrient[] | null | undefined;
    texture: NutritionOrder_Texture[] | null | undefined;
    fluidConsistencyType: CodeableConcept[] | null | undefined;
    instruction: string | null | undefined;
}
